graylog2-rest-client/src/main/java/org/graylog2/restclient/lib/ApiClientImpl.java

```diff
@@ -519,14 +523,22 @@ private void ensureAuthentication() {
             }
         }
 
+        private class RequestAndFuture {
+            private final Request request;
+            private final ListenableFuture<Response> listenableFuture;
+            public RequestAndFuture(Request request, ListenableFuture<Response> listenableFuture) {
+                this.request = request;
+                this.listenableFuture = listenableFuture;
+            }
+        }
         @Override
-        public Map<Node, T> executeOnAll() {
+        public Map<Node, T> executeOnAll() throws APIException {
             HashMap<Node, T> results = Maps.newHashMap();
             if (node == null && nodes == null) {
                 nodes = serverNodes.all();
             }
 
-            final Map<Node, ListenableFuture<Response>> requests = Maps.newHashMap();
+            final Map<Node, RequestAndFuture> requests = Maps.newHashMap();
```

Comment:
We know the final size of the map (`nodes.size()`) so we could initialize it with the correct size.